app/controllers/admin/orders_controller.rb

```diff
@@ -2,12 +2,31 @@
 
 module Admin
   class OrdersController < AdminController
+    before_action :set_order, only: %i[update]
+
     def index
       @pagy, @orders = pagy(Order.includes(%i[shipment payment]).order(created_at: :desc).decorate)
     end
 
     def show
       @order = Order.find(params[:id]).decorate
     end
+
+    def update
+      event = params[:event]
+
+      if Orders::StateService.new(@order).send(event)
+        flash[:success] = 'Order\'s status change has been succesfull'
+      else
+        flash[:alert] = 'Can\'t change order\'s status'
+      end
+      redirect_to admin_order_path(@order)
+    end
+
+    private
+
+    def set_order
```

Comment:
you use this also in a show method.
Btw. currently using callbacks are not popular

Reply:
I have not used that in the show method, because show is using decorator, and wanted to avoid something like @order=@order.decorate.
I have fixed that by removing callback (also in payments and shipments controllers)